nnsmith/materialize/tensorflow/__init__.py

```diff
@@ -222,4 +222,4 @@ class TFModelGPU(TFModel):
     def device(self) -> tf.device:
         gpus = tf.config.list_physical_devices("GPU")
         assert gpus, "No GPU available"
-        return tf.device(gpus[0].name)
+        return tf.device("/GPU:0")
```

Comment:
Just curious about the change. I thought using TF's public API `list_physical_devices` is more stable than hard-coded strings. Isn't it?

Reply:
Just as the second item in the PR message, directly using `gpus[0].name` which will be `'/physical_device:GPU:0'` at runtime will lead to an exception.

Reply:
Oh, I see, I think we should go `list_logical_devices` then? `list_logical_devices` should also be used in XLA IMO.

Reply:
You have a point.